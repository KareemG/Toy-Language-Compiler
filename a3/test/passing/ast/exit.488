
{
    var a : integer

    while true do
    {
        exit
    }

    while true do
    {
        while true do
        {
            read a

            if a = 1 then exit
            exit 2 when a = 2
        }
    }

    repeat
    {
        repeat
        {
            read a
            exit when a = 1
            if a = 2 then exit 2
        }
        until true
    }
    until true
}