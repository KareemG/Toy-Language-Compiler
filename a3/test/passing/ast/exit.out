 { 
 var a : integer
 while true do
 { 
 exit
 } 
 while true do
 { 
 while true do
 { 
 read a
 if (a) = (1) then
 exitend
 exit 2 when (a) = (2)
 } 
 } 
 repeat { 
 repeat { 
 read a
 exit when (a) = (1)
 if (a) = (2) then
 exit 2end
 } 
 until 
 true
 } 
 until 
 true
 }